### PR TITLE
simulator: add hot region cases.

### PR DESCRIPTION
**Pull request by disksing**

### CODE REVIEW

File: pkg/faketikv/cases/hot_read.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+			Available: 9 * gb,
+		})
+	}
+	var id idAllocator
+	id.setMaxID(5)
+	for i := 0; i < 500; i++ {
+		peers := []*metapb.Peer{
+			{Id: id.nextID(), StoreId: uint64(rand.Intn(5) + 1)},
+			{Id: id.nextID(), StoreId: uint64(rand.Intn(5) + 1)},
+			{Id: id.nextID(), StoreId: uint64(rand.Intn(5) + 1)},
```
Comment (nolouch) [inline pkg/faketikv/cases/hot_read.go]: seems two peers may be in same one store.

File: server/schedule/basic_cluster.go [hunk 2]
```diff
@@ -270,7 +267,7 @@ func (bc *BasicCluster) UpdateWriteStatCache(region *core.RegionInfo, hotRegionT
 func (bc *BasicCluster) UpdateReadStatus(region *core.RegionInfo) {
 	var ReadBytesPerSec uint64
 	v, isExist := bc.ReadStatistics.Peek(region.GetId())
-	if isExist {
+	if isExist && !Simulating { // When simulating, we can't calculate it using physical time.
```
Comment (nolouch) [inline server/schedule/basic_cluster.go]: the `UpdateWriteStatus` also need to add.

File: server/schedule/opts.go [hunk 3]
```diff
@@ -17,6 +17,10 @@ import (
 	"time"
 )
 
+// Simulating is an option to overpass the impact of accelerated time. Should
+// only turned on by the simulator.
+var Simulating bool
```
Comment (siddontang) [inline server/schedule/opts.go]: I don't like the way using a global var. Maybe you can add a function like IsSimulating instead.

Comment (disksing) [reply server/schedule/opts.go]: I'm not quite understand your suggestion. Where should the function be defined and where should we set it?

Comment (siddontang) [reply server/schedule/opts.go]: Maybe in the cluster, you set it in the simulator.

File: pkg/faketikv/cases/hot_read.go [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+			ID:        uint64(i),
+			Status:    metapb.StoreState_Up,
+			Capacity:  10 * gb,
+			Available: 9 * gb,
+		})
+	}
+	var id idAllocator
+	id.setMaxID(5)
+	for i := 0; i < 500; i++ {
+		storeIDs := rand.Perm(10)
```
Comment (Connor1996) [inline pkg/faketikv/cases/hot_read.go]: there are only five stores

File: server/schedule/opts.go (hunk 3 above)
Comment (disksing) [reply server/schedule/opts.go]: Hi @siddontang , I would like not to do it.
It is doable to add a flag to `server.Config` and then pass the flag all around, but will involve a lot of adjustments to the components that do not need to know whether it is simulating or not.
In fact, only the code in the `schedule` and `schedulers` needs to be concerned with whether it is simulating. And the effect is global, so I think it's fine and simple to use to global flag.

Comment (siddontang) [reply server/schedule/opts.go]: I see that you only use this in BasicCluster, is it hard to pass from outside?

**Pull request merged**